src/util.rs

```diff
@@ -431,6 +435,86 @@ pub fn compile_cdn_list(cdn_file_path: Option<String>) -> Vec<String> {
     return_vec
 }
 
+fn sc_extract_exact_timestamps(html_fragment: &Html) -> Result<ExtractedTimestamps> {
+    let exact_dt_selector = match Selector::parse("div > div[data-requests]") {
+        Ok(s) => s,
+        Err(_) => Err(DeriveDateError::SelectorError)?,
+    };
+
+    match html_fragment
+        .select(&exact_dt_selector)
+        .nth(0)
+        .ok_or(DeriveDateError::ScraperElementError)
+    {
+        Ok(d) => {
+            match d
+                .value()
+                .attr("data-requests")
+                .ok_or(DeriveDateError::ScraperAttributeError)
+            {
+                Ok(s) => {
+                    // Parse the clips_json into the struct StreamsChartsTwitchClip with serde_json
+                    let clips_payloads: Vec<StreamsChartsTwitchClip> =
+                        serde_json::from_str(&s.to_string()).unwrap();
+                    let start_dt =
+                        match parse_timestamp(&clips_payloads.first().unwrap().started_at) {
+                            Ok(dt) => dt,
+                            Err(e) => return Err(e)?,
+                        };
+                    let end_dt = match parse_timestamp(&clips_payloads.last().unwrap().ended_at) {
+                        Ok(dt) => dt,
+                        Err(e) => return Err(e)?,
+                    };
+                    info!(
+                        "Found exact timestamps for the stream. Started at {} and ended at {}.",
+                        start_dt, end_dt
+                    );
+                    return Ok(ExtractedTimestamps {
+                        processing_type: ProcessingType::Exact,
+                        start_timestamp: start_dt,
+                        end_timestamp: end_dt,
+                    });
+                }
+                Err(e) => return Err(e)?,
+            }
+        }
+        Err(e) => return Err(e)?,
+    };
+}
+
+fn sc_bruteforce_timestamps(html_fragment: &Html) -> Result<ExtractedTimestamps> {
+    let bruteforce_selector = match Selector::parse("time") {
+        Ok(s) => s,
+        Err(_) => Err(DeriveDateError::SelectorError)?,
+    };
+    let date_init = match html_fragment
+        .select(&bruteforce_selector)
+        .nth(0)
+        .ok_or(DeriveDateError::ScraperElementError)
+    {
+        Ok(d) => {
+            match d
+                .value()
+                .attr("datetime")
+                .ok_or(DeriveDateError::ScraperAttributeError)
+            {
+                Ok(s) => s.to_string(),
```

Comment:
I think I wrote this part, so this is my bad, haha, but same thing here as with the ```serde``` part, no need to convert ```s``` to String if we're gonna turn it back into &str later.